pkg/vmprovider/providers/vsphere2/vmlifecycle/bootstrap.go

```diff
@@ -319,3 +325,97 @@ func isLinuxGuest(guestID string) bool {
 
 	return strings.Contains(guestID, "Linux") || strings.Contains(guestID, "linux")
 }
+
+func logConfigSpec(
+	vmCtx context.VirtualMachineContextA2,
+	configSpec *vimTypes.VirtualMachineConfigSpec) {
```

Comment:
The `configSpec` should always be non-nil, so please change this to by-value, and update the Sanitize function accordingly.